modules/javafx.graphics/src/main/native-glass/mac/PlatformSupport.m

```diff
@@ -0,0 +1,281 @@
+    }
+
+    // Window colors
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.windowBackgroundColor" value:[NSColor windowBackgroundColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.windowFrameTextColor" value:[NSColor windowFrameTextColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.underPageBackgroundColor" value:[NSColor underPageBackgroundColor]];
+
+    // Highlights and shadows
+    if (MACOS_10_13) {
+        [PlatformSupport putColor:preferences key:"macOS.NSColor.findHighlightColor" value:[NSColor findHighlightColor]];
+    }
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.highlightColor" value:[NSColor highlightColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.shadowColor" value:[NSColor shadowColor]];
+
+    // Adaptable system colors
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemBlueColor" value:[NSColor systemBlueColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemBrownColor" value:[NSColor systemBrownColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemGrayColor" value:[NSColor systemGrayColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemGreenColor" value:[NSColor systemGreenColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemIndigoColor" value:[NSColor systemIndigoColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemOrangeColor" value:[NSColor systemOrangeColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemPinkColor" value:[NSColor systemPinkColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemPurpleColor" value:[NSColor systemPurpleColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemRedColor" value:[NSColor systemRedColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemTealColor" value:[NSColor systemTealColor]];
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemYellowColor" value:[NSColor systemYellowColor]];
+}
+
++ (void)updatePreferences:(jobject)application {
+    GET_MAIN_JENV;
+
+    jobject newPreferences = [self collectPreferences];
+    if (newPreferences == nil) {
+        return;
+    }
+
+    jboolean preferencesChanged = !(*env)->CallBooleanMethod(
+        env, newPreferences, jObjectEqualsMethod, currentPreferences);
+    GLASS_CHECK_EXCEPTION(env);
+
+    if (preferencesChanged) {
+        if (currentPreferences != nil) {
+            (*env)->DeleteGlobalRef(env, currentPreferences);
+        }
+
+        currentPreferences = (*env)->NewGlobalRef(env, newPreferences);
+
+        jobject unmodifiablePreferences = (*env)->CallStaticObjectMethod(
+            env, jCollectionsClass, jCollectionsUnmodifiableMapMethod, newPreferences);
+        GLASS_CHECK_EXCEPTION(env);
+
+        if (unmodifiablePreferences != nil) {
+            (*env)->CallVoidMethod(
```

Comment:
I think you should call `GLASS_CHECK_EXCEPTION` here or at the end of the method to clear any pending exception.